pkg/skaffold/event/event.go

```diff
@@ -739,6 +736,7 @@ func InititializationFailed(err error) {
 			},
 		},
 	})
+	handler.handle(nil)
```

Comment:
A `nil` event seems really odd — if it indicates that the connection is being closed, why not a Closed event?  IIUC a nil event is not sent on normal termination?

Worth adding a comment as to why this is required.

Reply:
Can you take a look now.  I have added some logic to close the event channel there by indicating all receivers that no more events are coming.